// Copyright 2023 Ant Group Co., Ltd.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// 黑盒互联互通传输层协议定义
// Black box interconnection transport layer protocol definition

// this proto only for TEST

syntax = "proto3";

package blackbox_interconnect;

option cc_generic_services = true;

message HttpRequest {};
message HttpResponse {};

service DummyBlackBoxService {
  rpc default_method(HttpRequest) returns (HttpResponse);
}

// request uri:
// topic will be in httpheaders
// /v1/interconn/chan/pop?timeout=10
// /v1/interconn/chan/push

// source node will use
// /v1/interconn/chan/invoke
// to push message to target node
